src/mon/OSDMonitor.cc

```diff
@@ -1230,6 +1033,180 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   encode_health(next, t);
 }
 
+void OSDMonitor::encode_pending_sub_op(OSDMap& tmp) {
+  tmp.deepish_copy_from(osdmap);
+  tmp.apply_incremental(pending_inc);
+
+  {
+    if (tmp.test_flag(CEPH_OSDMAP_FULL | CEPH_OSDMAP_NEARFULL)) {
+      dout(10) << __func__ << " clearing legacy osdmap nearfull/full flag"
+               << dendl;
+      remove_flag(CEPH_OSDMAP_NEARFULL);
+      remove_flag(CEPH_OSDMAP_FULL);
+    }
+  }
+
+  std::set<int64_t> full_pool_ids;
+  std::set<int64_t> backfillfull_pool_ids;
+  std::set<int64_t> nearfull_pool_ids;
+  tmp.get_full_pools(cct, &full_pool_ids, &backfillfull_pool_ids,
+                     &nearfull_pool_ids);
+
+  // Empty!
+  if (full_pool_ids.empty() ||
+      backfillfull_pool_ids.empty() ||
+      nearfull_pool_ids.empty()) {
+    // normal case - no nearfull, backfillfull or full osds try cancel any 
+    // improper nearfull/backfillfull/full pool flags first
+    for (auto &pool : tmp.get_pools()) {
+      auto p = pool.first;
+      if (tmp.get_pg_pool(p)->has_flag(pg_pool_t::FLAG_NEARFULL) &&
+          nearfull_pool_ids.empty()) {
+        dout(10) << __func__ << " clearing pool '" << tmp.pool_name[p]
+                 << "'s nearfull flag" << dendl;
+        if (pending_inc.new_pools.count(p) == 0) {
+          pending_inc.new_pools[p] = pool.second;
+        }
+        pending_inc.new_pools[p].flags &= ~pg_pool_t::FLAG_BACKFILLFULL;
```

Comment:
what happened here? This should have been `FLAG_NEARFULL`, and instead is `FLAG_BACKFILLFULL`, whereas the `if case` for backfillfull is simply gone. I don't think those two things are the same, so this feels a log like a copy-paste error.